control/autonomous_emergency_braking/src/node.cpp

```diff
@@ -357,9 +357,27 @@ bool AEB::checkCollision(MarkerArray & debug_markers)
 bool AEB::hasCollision(
   const double current_v, const Path & ego_path, const std::vector<ObjectData> & objects)
 {
+  if (ego_path.empty()) {
+    RCLCPP_ERROR_STREAM(get_logger(), "[AEB] ego_path is empty");
+    return false;
+  }
+  double current_x = 0.0;
+  double current_y = 0.0;
+  double current_yaw = 0.0;
+  const double current_w = angular_velocity_ptr_->z;
   // calculate RSS
-  const auto current_p = tier4_autoware_utils::createPoint(
-    ego_path[0].position.x, ego_path[0].position.y, ego_path[0].position.z);
+  geometry_msgs::msg::Pose current_p;
+  constexpr double epsilon = 1e-6;
+  const double & dt = imu_prediction_time_interval_;
+  const double & horizon = imu_prediction_time_horizon_;
+  for (double t = 0.0; t < horizon + epsilon; t += dt) {
+    current_x = current_x + current_v * std::cos(current_yaw) * dt;
+    current_y = current_y + current_v * std::sin(current_yaw) * dt;
+    current_yaw = current_yaw + current_w * dt;
+
+    current_p.position = tier4_autoware_utils::createPoint(current_x, current_y, 0.0);
+    current_p.orientation = tier4_autoware_utils::createQuaternionFromYaw(current_yaw);
+  }
```

Comment:
Thank you for the PR :+1: ! I believe there's no need for an implementation like this. This function is invoked when using the calculated predicted path (output of control) without relying on the IMU. Additionally, AEB can check for collisions with either the IMU or Control. Therefore, I think it's enough to:
```cpp
  if (ego_path.empty()) {
    RCLCPP_ERROR_STREAM(get_logger(), "[AEB] ego_path is empty");
    return false;
  }
  ``` 
  This situation occurs when ego goes off lanelet2 map.

Reply:
What I want to do here is to minimize the effect of ego_path on the calculation of other variables, and even if the path is not generated, EGO will still have a current pose. Of course, this current pose will not be useful for any calculations.

Reply:
@tkimura4-san Can you check this PR when you available? Thanks in advance

Reply:
@ismetatabay 
Sorry for the delayed response.

> I believe there's no need for an implementation like this.
 I agree with this.